EssentialFeed/EssentialFeedTests/Feed API/LoadFeedImageCommentsFromRemoteUseCaseTests.swift

```diff
@@ -0,0 +1,225 @@
+			})
+		}
+	}
+	
+	func test_cancelLoadImageCommentsURLTask_cancelsClientURLRequest() {
+		let image = uniqueImage()
+		let url = URL(string: "https://a-given-url.com/image/\(image.id.uuidString)/comments")!
+		let (sut, client) = makeSUT(image: image)
+		
+		let task = sut.load() { _ in }
+		XCTAssertTrue(client.cancelledURLs.isEmpty, "Expected no cancelled URL request until task is cancelled")
+		
+		task.cancel()
+		XCTAssertEqual(client.cancelledURLs, [url], "Expected cancelled URL request after task is cancelled")
+	}
+	
+	func test_loadImageCommentsFromURL_doesNotDeliverResultAfterCancellingTask() {
+		let (sut, client) = makeSUT()
+		let nonEmptyData = Data("non-empty data".utf8)
+		
+		var received = [FeedImageCommentsLoader.Result]()
+		let task = sut.load() { received.append($0) }
+		task.cancel()
+		
+		client.complete(withStatusCode: 404, data: anyData())
+		client.complete(withStatusCode: 200, data: nonEmptyData)
+		client.complete(with: anyNSError())
+		
+		XCTAssertTrue(received.isEmpty, "Expected no received results after cancelling task")
+	}
+	
+	func test_loadImageCommentsFromURL_doesNotDeliverResultAfterSUTInstanceHasBeenDeallocated() {
+		let client = HTTPClientSpy()
+		let image = uniqueImage()
+		let url = URL(string: "https://a-given-url.com/image/\(image.id.uuidString)/comments")!
+		var sut: RemoteFeedImageCommentsLoader? = RemoteFeedImageCommentsLoader(url: url, client: client)
+		
+		var capturedResults = [FeedImageCommentsLoader.Result]()
+		_ = sut?.load { capturedResults.append($0) }
+		
+		sut = nil
+		client.complete(withStatusCode: 200, data: anyData())
+		
+		XCTAssertTrue(capturedResults.isEmpty)
+	}
+	
+	// MARK: - Helpers
+	
+	private func makeSUT(file: StaticString = #filePath, line: UInt = #line, image: FeedImage = uniqueImage()) -> (sut: RemoteFeedImageCommentsLoader, client: HTTPClientSpy) {
+		let client = HTTPClientSpy()
+		let url = URL(string: "https://a-given-url.com/image/\(image.id.uuidString)/comments")!
+		let sut = RemoteFeedImageCommentsLoader(url: url, client: client)
+		trackForMemoryLeaks(sut, file: file, line: line)
+		trackForMemoryLeaks(client, file: file, line: line)
+		return (sut, client)
+	}
+	
+	private func failure(_ error: RemoteFeedImageCommentsLoader.Error) -> FeedImageCommentsLoader.Result {
+		return .failure(error)
+	}
+	
+	private func makeItemsJSON(_ items: [[String: Any]]) -> Data {
+		let json = ["items": items]
+		return try! JSONSerialization.data(withJSONObject: json)
+	}
+	
+	private func makeItem(id: UUID, message: String, createdAt: (date: Date, iso8601String: String), username: String) -> (model: FeedImageComment, json: [String: Any]) {
+		let item = FeedImageComment(id: id, message: message, createdAt: createdAt.date, author: .init(username: username))
+		let json = [
+			"id": id.uuidString,
+			"message": message,
+			"created_at": createdAt.iso8601String,
+			"author": ["username": username]
+		].mapValues { $0 }
```

Comment:
`mapValues { $0 }` is not needed here. There's no mapping going on - you're always returning the same values ($0).

Remove it.

```suggestion
		let json: [String: Any] = [
			"id": id.uuidString,
			"message": message,
			"created_at": createdAt.iso8601String,
			"author": ["username": username]
		]
```